log/context.go

```diff
@@ -0,0 +1,62 @@
+package log
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+)
+
+type contextKeyType struct{}
+
+var contextKey contextKeyType
```

Comment:
OOC, why use a struct rather than just a typed int?

Reply:
an empty `struct{}` consumes 0 bytes of memory (plus type info), an `int` is 8 bytes (plus type info) :)